src/logger/mongoose_flatlog_formatter.erl

```diff
@@ -1,10 +1,35 @@
 -module(mongoose_flatlog_formatter).
 -export([format/2]).
 
-format(Map, UsrConfig) ->
+
+-spec format(logger:log_event(), logger:formatter_config()) -> unicode:chardata().
+format(Event, FConfig) ->
+    try do_format(Event, FConfig)
+    catch
+        %% Exceptions during log formatting can lead to a death spiral of recursive error logging, so
+        %% format the formatter error in a safe way and don't allow the exception to propagate.
+        Class:Reason -> format_log_formatter_error(Class, Reason, Event, FConfig)
```

Comment:
We prefer to catch with `catch error:Class:Stacktrace`.

- Don't ignore stacktrace when catching - it would pretty hard to debug without line information.
- ignore throw and exits.

Reply:
Thanks for the suggestion - added